app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -0,0 +1,193 @@
+        onSelectDelete: ((Int) -> Unit)?,
+        /**
+         * Opens a Dialog to Edit an object in the list.
+         */
+        onSelectEdit: ((Int?) -> Unit)?,
+        /**
+         * Called when the Dialog Completes through any listener.
+         * This will reset all values to null in preparation for the next opening of the Dialog.
+         */
+        onDialogComplete: (() -> Unit)?,
+    ): ((Int?, String) -> Unit) {
+        val sheet = BottomSheetDialog(context)
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.recycler_layer_list_layout, null)
+        val data = mutableListOf<AdapterViewModel>()
+        view.findViewById<Button>(R.id.closeLayerListView).setOnClickListener {
+            onDialogComplete?.invoke()
+            sheet.dismiss()
+        }
+        (layerList.size - 1 downTo 0).forEach {
+            val layerListNameAtIndex = layerList[it]
+            data.add(
+                data.size, LayerViewModel(layerName = layerListNameAtIndex,
+                    onDeleteLayer = { onSelectDelete?.invoke(it) },
+                    onEditName = { onSelectEdit?.invoke(it) })
+            )
+        }
+        val layerListRecycler = view.findViewById<RecyclerView>(R.id.listOfLayers)
+        layerListRecycler.apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = RecyclerViewAdapter(
+                data = data,
+            )
+        }
+        sheet.setOnDismissListener {
+            onDialogComplete?.invoke()
+            sheet.dismiss()
+        }
+        sheet.setContentView(view)
+        sheet.show()
+        return { index, newName ->
+            if (index != null) {
+                data[index] = LayerViewModel(layerName = newName,
+                    onDeleteLayer = { onSelectDelete?.invoke(index) },
+                    onEditName = { onSelectEdit?.invoke(index) })
+            }
+            layerListRecycler.adapter?.notifyItemChanged(index ?: 0, newName)
+        }
+    }
+
+    /**
+     * Function that will search for if the default hint is correct or needs to be changed.
+     */
+    private fun createLayerHint(context: Context, layerList: MutableList<String>): String {
+        var currentHint = context.getString(R.string.layerHint, hintIndex)
+        while (layerList.contains(currentHint)) {
+            hintIndex += 1
+            currentHint = context.getString(R.string.layerHint, hintIndex)
+        }
+        hintIndex = LAYER_COUNT_START
+        return currentHint
+    }
+
+    /**
+     * Function that will return a Pair of a String and a Boolean.
+     * If there is a match for the String in the list, or if the String is too long, it will return an error message and true.
+     * If there is no errors, it will return an empty String and False.
+     */
+    private fun checkForErrors(
+        context: Context, layerList: MutableList<String>, newLayerName: String
+    ): Pair<String, Boolean> {
```

Comment:
Let's not return a pair when we can avoid it. In general, opt to use a data class instead of a pair. They're more type safe.
In this case, can't you just return a nullable string? If null, no error, not null = error?